go/vt/mysqlctl/backupengine.go

```diff
@@ -213,8 +219,14 @@ func registerBackupEngineFlags(fs *pflag.FlagSet) {
 // a particular backup by calling GetRestoreEngine().
 //
 // This must only be called after flags have been parsed.
-func GetBackupEngine() (BackupEngine, error) {
-	name := backupEngineImplementation
+func GetBackupEngine(backupEngine *string) (BackupEngine, error) {
```

Comment:
Let's never pass a `*string`. This function should accept a `string`. If it's empty, use the default value.

Reply:
the reason it is a `*string` is because that's how grpc optional arguments are created (as it should avoid backward compatibility issues). just to confirm, you prefer me to get rid of it completely and make it mandatory or just avoid passing a `*string` to `GetBackupEngine()`? I will keep the grpc optional parameter for now, but let me know if you prefer the other way

Reply:
> or just avoid passing a *string to GetBackupEngine()

This.